lib/viewing/ViewsFromServer.js

```diff
@@ -0,0 +1,48 @@
+"use strict";
+
+var $ = require("../application/jquery");
+
+var IDENTITY_ATTRIBUTE = "data-view-uid";
+
+var viewElements = {};
+
+var ViewsFromServer = {
+
+    IDENTITY_ATTRIBUTE: IDENTITY_ATTRIBUTE,
+
+    initialize: function() {
+        var $viewElements = $("[" + IDENTITY_ATTRIBUTE + "]");
+
+        $viewElements.each(function(i, viewElement) {
+            var uid = $(viewElement).attr(IDENTITY_ATTRIBUTE);
+
+            viewElements[uid] = viewElement;
+        });
+    },
```

Comment:
right - the gain here is algorithmic. as @ericherdzik points out, the difference is 1 $ attribute query vs n where n is the number of views. the attribute selector query is one of the slowest jquery selectors (http://jsperf.com/id-vs-class-vs-tag-selectors/2).

After the the 1st lookup in the ClientApp, every time reattach is called by a View, it does a constant lookup into the viewElements hash rather than another attribute selector lookup through the whole DOM. as the number of Views on a page goes up, the gain would be more noticeable. 

all that said, i can see and feel a difference in chrome. with this change, i can see events that happen onDOM happen closer together. i can also feel it when i scroll down onload - there is less jank as i scroll.